quiz.rb

```diff
@@ -14,24 +14,56 @@
 # Take a string and return an array of normalized words.
 # normalized words contain no punctuation and are all lower case
 def normalize_words words
+  return words.downcase.gsub(/[^a-z ]/, '').split(' ')
 end
 
 # Question 2
 # Take a string and return the count of normalized words
 def word_count words
+  words.downcase.gsub(/[^a-z ]/, '').split(' ')
+  return words.length
 end
 
 # Question 3
 # Take a string and return and array of the unique normalized words.
 def unique_words words
+  wordHash = {}
+  wordArr = words.downcase.gsub(/[^a-z ]/, '').split(' ')
+  i = 0;
+  while (i < wordArr.length)
+    i += 1
+  wordHash[wordArr[i]] = true
+  end
+  return wordHash
 end
 
 # Question 4
 # Take a string and return the count of unique normalized words.
 def unique_word_count words
+  wordHash = {}
+  wordArr = words.downcase.gsub(/[^a-z ]/, '').split(' ')
+  count = 0
+  i = 0;
+  while (i < wordArr.length)
+  i += 1
+  wordHash[wordArr[i]] = true
+  end
+  return count
 end
 
 # Question 5
 # Take a string and return a Hash of the nomalized words as keys and the count for the words as the values
 def word_frequencies words
+  wordHash = {}
+  wordArr = words.downcase.gsub(/[^a-z ]/, '').split(' ')
+  count = 0;
+
+  i = 0;
+  while (i < wordArr.length)
```

Comment:
looping was a big part of javascript, but in ruby, we have enumerators that can do the same work much more efficiently. we'll get more into these as we dive deeper into ruby